chainerx_cc/chainerx/native/native_device/misc.cc

```diff
@@ -7,51 +7,27 @@
 #include "chainerx/device.h"
 #include "chainerx/dtype.h"
 #include "chainerx/kernels/math.h"
+#include "chainerx/kernels/misc.h"
 #include "chainerx/native/elementwise.h"
 #include "chainerx/native/kernel_regist.h"
 #include "chainerx/numeric.h"
-#include "chainerx/routines/math.h"
+#include "chainerx/routines/type_util.h"
 
 namespace chainerx {
 namespace native {
 namespace {
 
-CHAINERX_NATIVE_REGISTER_ELTWISE_FLOAT_UNARY_KERNEL(FabsKernel, { out = chainerx::Fabs(x); });
+CHAINERX_NATIVE_REGISTER_ELTWISE_FLOAT_UNARY_KERNEL(SqrtKernel, { out = chainerx::Sqrt(x); });
 
-CHAINERX_NATIVE_REGISTER_ELTWISE_DTYPE_UNARY_KERNEL(SignKernel, { out = chainerx::Sign(x); }, VisitNumericDtype);
+CHAINERX_NATIVE_REGISTER_ELTWISE_FLOAT_UNARY_KERNEL(SquareKernel, { out = x * x; });
 
-class NativeSquareKernel : public SquareKernel {
-public:
-    void Call(const Array& x, const Array& out) override {
-        x.device().CheckDevicesCompatible(x, out);
-        VisitFloatingPointDtype(out.dtype(), [&](auto pt) {
-            using T = typename decltype(pt)::type;
-            struct Impl {
-                void operator()(int64_t /*i*/, T x, T& out) { out = x * x; }
-            };
-            Elementwise<const T, T>(Impl{}, x, out);
-        });
-    }
-};
+CHAINERX_NATIVE_REGISTER_ELTWISE_FLOAT_UNARY_KERNEL(FabsKernel, { out = chainerx::Fabs(x); });
 
-CHAINERX_NATIVE_REGISTER_KERNEL(SquareKernel, NativeSquareKernel);
+CHAINERX_NATIVE_REGISTER_ELTWISE_DTYPE_UNARY_KERNEL(SignKernel, { out = chainerx::Sign(x); }, VisitNumericDtype);
 
-class NativeSqrtKernel : public SqrtKernel {
-public:
-    void Call(const Array& x, const Array& out) override {
-        x.device().CheckDevicesCompatible(x, out);
-        const Array& x_cast = x.dtype() == out.dtype() ? x : x.AsType(out.dtype());
-        VisitFloatingPointDtype(out.dtype(), [&](auto pt) {
-            using T = typename decltype(pt)::type;
-            struct Impl {
-                void operator()(int64_t /*i*/, T x, T& out) { out = chainerx::Sqrt(x); }
-            };
-            Elementwise<const T, T>(Impl{}, x_cast, out);
-        });
-    }
-};
+CHAINERX_NATIVE_REGISTER_ELTWISE_FLOAT_UNARY_KERNEL(CeilKernel, { out = chainerx::Ceil(x); });
```

Comment:
I think `ceil` and `floor` should be in `rounding`.

Reply:
I'll work on it in another PR.

Reply:
Then why did you move this in this PR?

Reply:
These kernels were defined here before this PR. This PR just rewrites the definitions of kernels in `*_device/misc.cc` using macro. Should we revert them?

Reply:
OK, I understand, but it was difficult to notice in the review. Maybe it's better to separate the fix, or at least leave some comment.